operators/src/processing/point_in_polygon/tester.rs

```diff
@@ -0,0 +1,340 @@
+                    || (ring_number > 0 && is_coordinate_in_ring)
+                {
+                    // coordinate is either "not in outer ring" or "in inner ring"
+                    is_coordinate_in_polygon = false;
+                    break;
+                }
+            }
+
+            if is_coordinate_in_polygon {
+                is_coordinate_in_multi_polygon = true;
+                break;
+            }
+        }
+
+        is_coordinate_in_multi_polygon
+    }
+
+    fn coordinate_in_multi_polygon_iter<'p>(
+        &'p self,
+        coordinate: &'p Coordinate2D,
+        time_interval: &'p TimeInterval,
+    ) -> impl Iterator<Item = bool> + 'p {
+        let polygon_offsets = self.polygon_offsets;
+        let ring_offsets = self.ring_offsets;
+
+        let time_intervals = self.time_intervals;
+
+        two_tuple_windows(self.feature_offsets.iter().map(|&c| c as usize))
+            .zip(time_intervals)
+            .map(
+                move |(
+                    (multi_polygon_start_index, multi_polygon_end_index),
+                    multi_polygon_time_interval,
+                )| {
+                    if !multi_polygon_time_interval.intersects(time_interval) {
+                        return false;
+                    }
+
+                    self.check_coordinate_in_multipolygons(
+                        coordinate,
+                        polygon_offsets,
+                        ring_offsets,
+                        multi_polygon_start_index,
+                        multi_polygon_end_index,
+                    )
+                },
+            )
+    }
+
+    /// Is the coordinate contained in any polygon of the collection?
+    ///
+    /// The function returns `true` if the `Coordinate2D` is inside the multi polygon, or
+    /// `false` if it is not. If the point is exactly on the edge of the polygon,
+    /// then the function may return `true` or `false`.
+    ///
+    /// TODO: check boundary conditions separately
+    ///
+    pub fn is_coordinate_in_any_polygon(
+        &self,
+        coordinate: &Coordinate2D,
+        time_interval: &TimeInterval,
+    ) -> bool {
+        self.coordinate_in_multi_polygon_iter(coordinate, time_interval)
+            .any(std::convert::identity)
+    }
+
+    #[allow(dead_code)]
+    pub fn multi_polygons_containing_coordinate(
+        &self,
+        coordinate: &Coordinate2D,
+        time_interval: &TimeInterval,
+    ) -> Vec<bool> {
+        self.coordinate_in_multi_polygon_iter(coordinate, time_interval)
+            .collect()
+    }
+}
+
+/// Loop through an iterator by yielding the current and previous tuple. Starts with the
+/// (first, second) item, so the iterator must have more than one item to create an output.
+fn two_tuple_windows<I, T>(mut iter: I) -> impl Iterator<Item = (T, T)>
+where
+    I: Iterator<Item = T>,
+    T: Copy,
+{
+    let mut last = iter.next();
+
+    iter.map(move |item| {
+        let output = (last.unwrap(), item);
```

Comment:
I am not sure. This is not my code and I didn't touch it in the PR. Maybe ask Chris.

Reply:
Hm, as far as I see there are two options:

1. `last` (255) is `Some`.
Then,  `unwrap` works the first time, only `Some` will be stored and it will work the next time.
2. `last` is `None`.
The iterator is empty so the function in `map` won't be ever called. The returned iterator is empty.

So it seems fine.